src/RA_Core.cpp

```diff
@@ -1310,26 +1324,37 @@ BOOL _ReadTil(const char nChar, char* buffer, unsigned int nSize, DWORD* pCharsR
 
 char* _ReadStringTil(char nChar, char* restrict& pOffsetInOut, BOOL bTerminate) noexcept
 {
+    Expects(pOffsetInOut != nullptr);
     char* pStartString = pOffsetInOut;
 
     while ((*pOffsetInOut) != '\0' && (*pOffsetInOut) != nChar)
+    {
         pOffsetInOut++;
-
+        Ensures(pOffsetInOut != nullptr);
```

Comment:
This will only be true if `pOffsetInOut` is `0xFFFFFFFF` before it's incremented. Is the tool telling you to add all of these things? I don't like it. I'm fine with null assertions on input parameters, or function returns, but throwing these around (especially when it's clear they'll never be false) is just polluting the code.

Reply:
Additionally, `assert` is compiled out of a release build. Terminating the app is arguably less useful than throwing an uncaught exception. At least with an exception you have some idea why the app terminated.

Reply:
exceptions are enabled